records_mover/records/types.py

```diff
@@ -18,150 +20,108 @@
-        meta: RecordsManifestEntryMetadata
+class LegacyRecordsManifestEntry(BaseRecordsManifestEntry, total=False):
+    meta: RecordsManifestEntryMetadata
 
-    class RecordsManifestEntryWithLength(BaseRecordsManifestEntry):
-        meta: RecordsManifestEntryMetadata
 
-    class LegacyRecordsManifest(TypedDict):
-        entries: List[LegacyRecordsManifestEntry]
+class RecordsManifestEntryWithLength(BaseRecordsManifestEntry):
+    meta: RecordsManifestEntryMetadata
 
-    class RecordsManifestWithLength(TypedDict):
-        entries: List[RecordsManifestEntryWithLength]
 
-    class UrlDetailsEntry(TypedDict):
-        content_length: int
+class LegacyRecordsManifest(TypedDict):
+    entries: List[LegacyRecordsManifestEntry]
 
-    Url = str
-    UrlDetails = Dict[Url, UrlDetailsEntry]
 
-    from typing_extensions import Literal  # noqa
+class RecordsManifestWithLength(TypedDict):
+    entries: List[RecordsManifestEntryWithLength]
 
-    from mypy_extensions import TypedDict
 
-    #
-    # Note: Any expansion of these types should also be done in
-    # records.jobs.hints
-    #
-    RecordsFormatType = Literal['delimited', 'parquet']
+class UrlDetailsEntry(TypedDict):
+    content_length: int
 
-    DelimitedVariant = Literal['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
 
-else:
-    RecordsManifestEntryMetadata = Mapping[str, int]
-    LegacyRecordsManifestEntry = Mapping[str, Union[str, bool, int, RecordsManifestEntryMetadata]]
-    RecordsManifestEntryWithLength =\
-        Mapping[str, Union[str, bool, int, RecordsManifestEntryMetadata]]
-    LegacyRecordsManifest = Mapping[str, List[LegacyRecordsManifestEntry]]
-    RecordsManifestWithLength = Mapping[str, List[RecordsManifestEntryWithLength]]
-    UrlDetailsEntry = Dict[str, int]
-    Url = str
-    UrlDetails = Dict[Url, UrlDetailsEntry]
+Url = str
+UrlDetails = Dict[Url, UrlDetailsEntry]
 
-    RecordsFormatType = str
 
-    DelimitedVariant = str
+#
+# Note: Any expansion of these types should also be done in
+# records.jobs.hints
+#
+RecordsFormatType = Literal['delimited', 'parquet']
 
-if TYPE_CHECKING:
-    HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
-                           "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
-else:
-    HintEncoding = str
+VALID_VARIANTS = ['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
+DelimitedVariant = Literal['dumb', 'csv', 'bigquery', 'bluelabs', 'vertica']
+
+HintEncoding = Literal["UTF8", "UTF16", "UTF16LE", "UTF16BE",
+                       "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"]
 
 VALID_ENCODING: List[HintEncoding] = [
     "UTF8", "UTF16", "UTF16LE", "UTF16BE", "UTF16BOM", "UTF8BOM", "LATIN1", "CP1252"
 ]
 
-if TYPE_CHECKING:
-    HintQuoting = Literal["all", "minimal", "nonnumeric", None]
-else:
-    HintQuoting = Optional[str]
+HintQuoting = Literal["all", "minimal", "nonnumeric", None]
 
 VALID_QUOTING: List[HintQuoting] = ["all", "minimal", "nonnumeric", None]
 
-
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker